File: TasteITServer/src/main/java/com/example/testeitserver/model/Post.java
package com.example.testeitserver.model;

import com.example.testeitserver.model.enums.PostType;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Document
@Data
public class Post {
    @Id
    private String postId;
    private String userId;
    private PostType postType;
    private PostMedia postMedia;
    private Recipe recipe;
    @DBRef
    private List<Tag> tags = new ArrayList<>();
    private Date date;
    @DBRef
    private List<Like> likes = new ArrayList<>();
    @DBRef
    private List<Comment> comments = new ArrayList<>();
}

File: MixITApp/src/app/model/post/Post.ts
import {PostMedia} from "./PostMedia";
import {Recipe} from "./Recipe";
import {Tag} from "../user/Tag";
import {Like} from "./Like";
import {EPostType} from "./EPostType";
import {PostAuthor} from "./PostAuthor";

export interface Post {
  postId?: string;
  postAuthorDto?: PostAuthor
  userId?: string;
  postType?: EPostType;
  postMedia?: PostMedia;
  tags?: Tag[];
  createdDate?: Date;
  likesCount?: number;
  recipe?: Recipe;
  commentsCount?: number;
  likedByCurrentUser?: boolean;

}
